src/components/dialog/Dialog.spec.js

```diff
@@ -69,6 +69,38 @@ describe('BDialog', () => {
         expect(wrapper.vm.onConfirm).toHaveBeenCalled()
     })
 
+    it('async confirm and keep Loading state', async () => {
+        wrapper.setProps({
+            onConfirm: jest.fn((confirmValue, { startLoading }) => {
+                startLoading()
+                expect(wrapper.vm.isLoading).toBeTruthy()
+                return new Promise(() => {})
+            }),
+            closeOnConfirm: false
+        })
+        expect(wrapper.vm.isLoading).toBeFalsy()
+        await wrapper.vm.confirm()
+        expect(wrapper.vm.isLoading).toBeTruthy()
+        expect(wrapper.vm.onConfirm).toHaveBeenCalled()
+    })
+
+    it('async confirm and close Loading state', async () => {
+        wrapper.setProps({
+            onConfirm: jest.fn((confirmValue, { startLoading, cancelLoading }) => {
+                startLoading()
+                expect(wrapper.vm.isLoading).toBeTruthy()
+                return new Promise(() => {
+                    cancelLoading()
+                })
+            }),
+            closeOnConfirm: false
+        })
+        expect(wrapper.vm.isLoading).toBeFalsy()
+        await wrapper.vm.confirm()
```

Comment:
This `await` has no effect, because `confirm` method is "sync" and returns nothing.